### PR TITLE
softether module: fix dataDir option

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Closes https://github.com/NixOS/nixpkgs/issues/35339
cc @rick68

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by fpletz — APPROVED** [review]
So much nicer! Thanks!

**Comment by Mic92** [comment]
@GrahamcOfBorg build softether

**Comment by Mic92** [comment]
Softether uses a `-m64`, which is a x86-only option. Removing this could be an easy patch, but there might be other parts that do not work out of the box. I would disable aarch64 support in the latter case.

File: pkgs/servers/softether/4.18.nix [hunk 1]
```diff
@@ -48,6 +48,6 @@ stdenv.mkDerivation rec {
     homepage = https://www.softether.org/;
     license = licenses.gpl2;
     maintainers = [ maintainers.rick68 ];
-    platforms = platforms.linux;
+    platforms = filter (p: p != "aarch64-linux") platforms.linux;
```
Comment (Mic92) [inline pkgs/servers/softether/4.18.nix]: The compiler options restricts softether to exactly `x86_64-linux`. so just put `platforms = ['x86_64-linux'];`

Comment (Mic92) [reply pkgs/servers/softether/4.18.nix]: I was not precise when I said disabling `aarch64`.

Comment (gnidorah) [reply pkgs/servers/softether/4.18.nix]: @Mic92 x86 is supported. They use very rudimentary "configure" script:
`cp src/makefiles/${OS}_${CPU}.mak Makefile`
to pick suitable Makefile.

Comment (gnidorah) [reply pkgs/servers/softether/4.18.nix]: @Mic92 Thanks, aarch64 is disabled now, as I have no aarch64 machines to test. Also they don't support if officially yet: https://www.softether.org/4-docs/1-manual/5._SoftEther_VPN_Bridge_Manual/5.1_Operating_System_Requirements#5.1.2._Support_for_Linux

**Comment by gnidorah** [comment]
@fpletz @Mic92 Thanks!

**Pull request merged**